app/public/progressBar.js

```diff
@@ -0,0 +1,47 @@
+const PROGRESS_BAR = ((progressBar) => {
+    let width = 1;
+    let interval = null;
+    let tickrate = 30;
+    let timeTaken = 0;
+
+    function pushBar() {
+        if (width === 100) {
+            stopBar();
+        }
+        else {
+            width++;
+            progressBar.style.width = width + "%";
+            if (width === 33 || width === 66 || width === 100) {
```

Comment:
This is really strange. You're making an assumption that the bar will take 3 seconds. Why not track timeTaken as a float value instead of an integer? Then you could floor this value when showing it as text.